qiita_db/sql_connection.py

```diff
@@ -460,176 +397,199 @@ def execute_fetchall(self, sql, sql_args=None):
+                        r_idx = int(placeholder.group(2))
+                        # Value index
+                        v_idx = int(placeholder.group(3))
+                        try:
+                            sql_args[pos] = self._results[q_idx][r_idx][v_idx]
+                        except IndexError:
+                            # A previous query that was expected to retrieve
+                            # some data from the DB did not return as many
+                            # values as expected
+                            self._raise_execution_error(
+                                sql, sql_args,
+                                "The placeholder {%d:%d:%d} does not match to "
+                                "any previous result"
+                                % (q_idx, r_idx, v_idx))
+                        except TypeError:
+                            # The query that the placeholder is pointing to
+                            # is not expected to retrieve any value
+                            # (e.g. an INSERT w/o RETURNING clause)
+                            self._raise_execution_error(
+                                sql, sql_args,
+                                "The placeholder {%d:%d:%d} is referring to "
+                                "an SQL query that does not retrieve data"
+                                % (q_idx, r_idx, v_idx))
+        return sql, sql_args
+
+    def add(self, sql, sql_args=None, many=False):
+        """Add an sql query to the transaction
+
+        If the current query needs a result of a previous query in the
+        transaction, a placeholder of the form '{#:#:#}' can be used. The first
+        number is the index of the previous SQL query in the transaction, the
+        second number is the row from that query result and the third number is
+        the index of the value within the query result row.
+        The placeholder will be replaced by the actual value at execution time.
 
         Parameters
         ----------
-        queue : str
-            name of queue adding to
         sql : str
-            sql command to run
-        sql_args : list, tuple or dict, optional
-            the arguments to fill sql command with
+            The sql query
+        sql_args : list of objects, optional
+            The arguments to the sql query
         many : bool, optional
-            Whether or not this should be treated as an executemany command.
-            Default False
+            Whether or not we should add the query multiple times to the
+            transaction
 
         Raises
         ------
-        KeyError
-            queue does not exist
-        """
-        if not self._check_queue_exists(queue):
-            raise KeyError("Queue '%s' does not exist" % queue)
+        TypeError
+            If `sql_args` is provided and is not a list
 
+        Notes
+        -----
+        If `many` is true, `sql_args` should be a list of lists, in which each
```

Comment:
This is confusingly worded. Something along the lines of "...each list in the list contains the parameters for one SQL query of the many."